### ISSUE TITLE
[0.20.4] URL missing single character with extra slash

### ISSUE THREAD

**Issue by pe8ter** [systemjs/systemjs#1557]
**Overview**

Trying to load a simple script with a default .js extension but the URL is missing a character and has an extra slash. This did not happen in 0.20.3, but broke in 0.20.4. Here is my config:

```javascript
SystemJS.config({
    packages: {
        '/': {
            defaultExtension: 'js',
        },
    },
});
```

And for completeness my import:

```javascript
SystemJS.import('foo');
```

**Expected result**

The script foo.js should load and execute.

**Actual result**

The script fails to load with 400 (Bad Request):

```
GET http://localhost:3000//oo.js 400 (Bad Request)
N @ fetch.js:32
(anonymous) @ instantiate.js:165
Uncaught (in promise) Error: Fetch error: 400 Bad Request
  Instantiating http://localhost:3000//oo.js
  Loading /foo.js
    at fetch.js:37
t @ common.js:83
(anonymous) @ loader-polyfill.js:70
```

Apologies for the noisy error dump, but you can at least see the mangled URL.

**Comment by pe8ter** [comment]
I found the change between 0.20.3 and 0.20.4 that is the root of the issue. In config.js there is a line to create a variable `pkgName`. [You can see here](https://github.com/systemjs/systemjs/blob/d1bdd5d150b2ce49ccb877bf8ec0858dbb7fd63a/src/config.js#L192) that the variable gets assigned then used. In the previous version there was [one more line](https://github.com/systemjs/systemjs/commit/d1bdd5d150b2ce49ccb877bf8ec0858dbb7fd63a#diff-9d9a6cd82f41984872a66a3ab0d440c4L193) to cut off the trailing slash.

Now that the package name keeps the trailing slash when we get to [finding the sub-path from the normalized URL](https://github.com/systemjs/systemjs/blob/ca6dfee95d09a159d0ea8bd3e205e1e737675d80/src/resolve.js#L290), `packageKey` is too long so one extra character from the sub-path gets removed. The sub-path for `http://localhost:3000/foo` becomes `oo` instead of `foo` and thus the errors above.

**Comment by guybedford** [comment]
@pe8ter thanks I've added this fix as you suggest in https://github.com/systemjs/systemjs/commit/647ccef64ab961e0cf34c0bd294dd781b14b61ae.

**Comment by guybedford** [comment]
Released in 0.20.5

**Comment by pe8ter** [comment]
Thanks!
